src/osd/scheduler/mClockScheduler.h

```diff
@@ -98,6 +108,7 @@ class mClockScheduler : public OpScheduler, md_config_obs_t {
   const bool is_rotational;
   const unsigned cutoff_priority;
   MonClient *monc;
+  PerfCounters *logger = nullptr;
```

Comment:
I think instead of declare nullptr to here please do it during constructor initialization.

Reply:
What are the disadvantages of using case-1 to initialize variables? @mohit84 

case-1:
```
class A {
    int *i = nullptr;
};
```

case-2:
```
class A {
    int *i;
    A() : i(nullptr) {}
};
```

Reply:
done

Reply:
AFAIK we should avoid initializing a pointer directly within a class definition instead we should do via constructor initialization.